src/index.d.ts

```diff
@@ -9,54 +9,56 @@ export interface RoutineTypes {
     FULFILL: string;
 }
 
-export type RoutineAction<T> = Action<T>;
-
-export type RoutineActionCreator<T> = (payload?: T) => RoutineAction<T>;
-
-export interface RoutineActionCreators {
-    trigger: RoutineActionCreator<any>;
-    request: RoutineActionCreator<any>;
-    success: RoutineActionCreator<any>;
-    failure: RoutineActionCreator<any>;
-    fulfill: RoutineActionCreator<any>;
+export interface RoutineActionCreators<TActionCreator> {
+    trigger: TActionCreator;
+    request: ActionFunctionAny<Action<any>>;
```

Comment:
To support payload / meta creators that may be passed as `createRoutine('TYPE', payloadCreator, metaCreator)` in non-`trigger` actions. `RoutineActionCreator` limited `metaCreator` to be only `(payload?: any)` while it can be `(...args: any[])` to handle `myRoutine(payload, some, thing)`